js/classes/entities/midground/weapon/sword.js

```diff
@@ -1,3 +1,201 @@
+/**
+ * A class that represents Chad's sword.
+ * 
+ * @author Trae Claar
+ * @author Nathan Hinthorne
+ */
 class Sword {
+    /**
+     * Constructor for a Sword which takes an initial type.
+     * 
+     * @param {number} type the type of the Sword (please use Sword.TYPE_1, .TYPE_2, etc.)
+     * @throws {Error} Will throw an error if type is not a Sword member type.
+     */
+    constructor(type) {
+        Sword.checkType(type);
 
+        this.pos = CHAD.pos;
+        this.isAttacking = false;
+        this.hasHit = false;
+        this.speed = 0;
+        this.offsetX = 0;
+        this.offsetDirX = 0;
+
+        this.setType(type);
+
+        this.loadAnimations();
+    };
+
+    /** The size of the Sword on the spritesheet. */
+    static get SIZE() {
+        return new Vector(40, 40);
+    };
+
+    /** The scale factor applied to the Sword when drawing. */
+    static get SCALE() {
+        return 3;
+    };
+
+    /** The size of the Sword after the scale factor is applied. */
+    static get SCALED_SIZE() {
+        return Vector.multiply(Sword.SIZE, Sword.SCALE);
+    };
+
+    /** The maximum distance the Sword can be from Chad while jabbing before it begins retracting. */
+    static get MAX_OFFSET() {
+        return 30;
+    }
+
+    /** The offset from Chad's y-position applied to the Sword's y-position. */
+    static get Y_OFFSET() {
+        return 10;
+    }
+
+    /** The spritesheet containing all the Sword types. */
+    static get SPRITESHEET() {
+        return "./sprites/swords.png";
+    };
+
+    /** How fast the Sword moves while jabbing. */
+    static get JAB_SPEED() {
+        return 300;
+    };
+
+    /** How fast the Sword moves while retracting. */
+    static get RETRACT_SPEED() {
```

Comment:
Is this in milliseconds? Or some other unit? 
It may be useful to either have a comment or add the unit to the function name.

```suggestion
    static get RETRACT_SPEED_MS() {
```

Reply:
I believe that this is in pixels/second, as with all of our entity speed properties. Indicating units in some way would definitely be good - thanks for the suggestion!